src/os/bluestore/BlueFS.h

```diff
@@ -363,9 +423,32 @@ class BlueFS {
       buffer_appender.append_zero(len);
     }
 
+    void append(uint64_t value) {
+      uint64_t l0 = get_buffer_length();
+      ceph_assert(l0 + sizeof(value) <= std::numeric_limits<unsigned>::max());
+      buffer_appender.append((const char*)&value, sizeof(value));
+    }
+
+    void append_hole(uint64_t len) {
+      buffer.append_hole(len);
+    }
+
+    void prepend(uint64_t value) {
+      bufferlist tail;
+      tail.append((const char*)&value, sizeof(value));
+      tail.claim_append(buffer);
+      std::swap(tail, buffer);
+    }
+
+    void remove_trailing(uint64_t amount) {
+      bufferlist bl;
+      bl.substr_of(buffer, 0, buffer.length() - amount);
```

Comment:
this looks like a no-op for now - local bl is assigned with a substring from buffer. And that's it - I can see no outer effect.

Reply:
forgot to remove it from previous design